packages/router/router.ts

```diff
@@ -3970,11 +4022,16 @@ function createClientSideRequest(
   history: History,
   location: string | Location,
   signal: AbortSignal,
+  initial?: boolean,
   submission?: Submission
 ): Request {
   let url = history.createURL(stripHashFromPath(location)).toString();
   let init: RequestInit = { signal };
 
+  if (initial) {
+    init.headers = { "X-Remix-Initial-Load": "yes" };
```

Comment:
Right now this is how I'm planning to let the Remix `clientLoader` know if it's an initial request or not - See https://github.com/remix-run/remix/discussions/7634#discussioncomment-7265014